### PR TITLE
[FEATURE] Add prop to set Custom Document ID for Firebase - Create Document action #6105

### PR DESCRIPTION
**Pull request by lcaresia**
### <samp>🤖 Generated by Copilot at fb48ae1</samp>

This pull request adds a new feature, improves the documentation, and updates the dependencies of the `firebase_admin_sdk` package and its actions and sources. The new feature allows users to create Firebase documents with custom IDs using the `create-document` action. The documentation of the `replicate-event-firestore` action is enhanced with a description and better formatting. The dependencies of the package and its components are updated to fix compatibility issues and use the latest versions. The `firebase_admin_sdk` package version is bumped to 0.0.6 and the versions of the actions and sources are also incremented accordingly.

### <samp>🤖 Generated by Copilot at fb48ae1</samp>

> _We're updating the `firebase_admin_sdk` package_
> _To fix the bugs and add some features_
> _Heave away, me hearties, heave away_
> _We'll make the actions work with custom IDs_

### <samp>🤖 Generated by Copilot at fb48ae1</samp>

*  Bump the version of the firebase_admin_sdk package to 0.0.6 and format the package.json file ([link](https://github.com/PipedreamHQ/pipedream/pull/6107/files?diff=unified&w=0#diff-ce474d6dd368020a2fea187b70e3d6465219afa51704a7b041bae0bfc3b84dc2L3-R3), [link](https://github.com/PipedreamHQ/pipedream/pull/6107/files?diff=unified&w=0#diff-ce474d6dd368020a2fea187b70e3d6465219afa51704a7b041bae0bfc3b84dc2L23-L22))
*  Add a customId prop to the create-document action and update the createDocument method in the app to use it ([link](https://github.com/PipedreamHQ/pipedream/pull/6107/files?diff=unified&w=0#diff-0553b3f835bdf03f59a3a6b03458a26315361648e56d16dbc6d8d7612a9882f0L8-R8), [link](https://github.com/PipedreamHQ/pipedream/pull/6107/files?diff=unified&w=0#diff-0553b3f835bdf03f59a3a6b03458a26315361648e56d16dbc6d8d7612a9882f0L27-R37), [link](https://github.com/PipedreamHQ/pipedream/pull/6107/files?diff=unified&w=0#diff-1776b1d9067e5a4b58b6d15747eaa5522bde0248be1c863a2205b3e66b44191cL209-R214))
*  Update the @types/node dependency to 17.0.45 for the firebase_admin_sdk package and all the actions and sources that use it ([link](https://github.com/PipedreamHQ/pipedream/pull/6107/files?diff=unified&w=0#diff-e36cd34e56210ac7c4b37df01fada78a8ae46e242983493d0c97385eabe8d087L8-R8), [link](https://github.com/PipedreamHQ/pipedream/pull/6107/files?diff=unified&w=0#diff-daffcb371a53f589f1fc9c4902301dcc8b4dfc58f63b84dab5fe6009ddb76078L8-R8), [link](https://github.com/PipedreamHQ/pipedream/pull/6107/files?diff=unified&w=0#diff-e6fb0ebcb12463728524da8267772d81fe9125b11f23fec99885f249fb3c46dcL8-R8), [link](https://github.com/PipedreamHQ/pipedream/pull/6107/files?diff=unified&w=0#diff-7e65c80c4f7a1041475a3c979fd2722bda35d6a3834e72d75ebfc934181b58f1L8-R8), [link](https://github.com/PipedreamHQ/pipedream/pull/6107/files?diff=unified&w=0#diff-b4cdccd4baedbce6740b81d5a5724b5264a488c92d34313f253b4ad5a455cf9cL8-R8), [link](https://github.com/PipedreamHQ/pipedream/pull/6107/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL8757-R8754), [link](https://github.com/PipedreamHQ/pipedream/pull/6107/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL8766-R8763), [link](https://github.com/PipedreamHQ/pipedream/pull/6107/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL8852-R8849), [link](https://github.com/PipedreamHQ/pipedream/pull/6107/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL8918-R8915), [link](https://github.com/PipedreamHQ/pipedream/pull/6107/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL8925-R8922), [link](https://github.com/PipedreamHQ/pipedream/pull/6107/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL8962-R8959), [link](https://github.com/PipedreamHQ/pipedream/pull/6107/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL9001-R8998), [link](https://github.com/PipedreamHQ/pipedream/pull/6107/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL9081-R9078), [link](https://github.com/PipedreamHQ/pipedream/pull/6107/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL9234-R9231), [link](https://github.com/PipedreamHQ/pipedream/pull/6107/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL11439-R11436), [link](https://github.com/PipedreamHQ/pipedream/pull/6107/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL11446-R11443), [link](https://github.com/PipedreamHQ/pipedream/pull/6107/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL11474-R11471), [link](https://github.com/PipedreamHQ/pipedream/pull/6107/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL11499-R11496), [link](https://github.com/PipedreamHQ/pipedream/pull/6107/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL11506-R11503), [link](https://github.com/PipedreamHQ/pipedream/pull/6107/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL11512-R11509), [link](https://github.com/PipedreamHQ/pipedream/pull/6107/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL11560-R11557), [link](https://github.com/PipedreamHQ/pipedream/pull/6107/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL11566-R11563), [link](https://github.com/PipedreamHQ/pipedream/pull/6107/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL11674-R11672), [link](https://github.com/PipedreamHQ/pipedream/pull/6107/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL11685-R11683), [link](https://github.com/PipedreamHQ/pipedream/pull/6107/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL11702-R11700), [link](https://github.com/PipedreamHQ/pipedream/pull/6107/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL11723-R11721), [link](https://github.com/PipedreamHQ/pipedream/pull/6107/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL11740-R11738), [link](https://github.com/PipedreamHQ/pipedream/pull/6107/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL11746-R11744), [link](https://github.com/PipedreamHQ/pipedream/pull/6107/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL11753-R11751), [link](https://github.com/PipedreamHQ/pipedream/pull/6107/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL11776-R11774), [link](https://github.com/PipedreamHQ/pipedream/pull/6107/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL17238-R17236), [link](https://github.com/PipedreamHQ/pipedream/pull/6107/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL17394-R17392), [link](https://github.com/PipedreamHQ/pipedream/pull/6107/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL17409-R17407), [link](https://github.com/PipedreamHQ/pipedream/pull/6107/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL17526-R17524), [link](https://github.com/PipedreamHQ/pipedream/pull/6107/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL17567-R17565), [link](https://github.com/PipedreamHQ/pipedream/pull/6107/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL17684-R17682), [link](https://github.com/PipedreamHQ/pipedream/pull/6107/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL17815-R17813), [link](https://github.com/PipedreamHQ/pipedream/pull/6107/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL17876-R17874), [link](https://github.com/PipedreamHQ/pipedream/pull/6107/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL17981-R17979), [link](https://github.com/PipedreamHQ/pipedream/pull/6107/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL18031-R18029), [link](https://github.com/PipedreamHQ/pipedream/pull/6107/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL18052-R18050), [link](https://github.com/PipedreamHQ/pipedream/pull/6107/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL21231-R21229), [link](https://github.com/PipedreamHQ/pipedream/pull/6107/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL21251-R21249))
*  Add a description to the replicate-event-firestore action and format the code ([link](https://github.com/PipedreamHQ/pipedream/pull/6107/files?diff=unified&w=0#diff-d047e0f37f4b69c4818a0a79f15454b22933b6cc913d358b460a834e3ccc7f8dL8-R9), [link](https://github.com/PipedreamHQ/pipedream/pull/6107/files?diff=unified&w=0#diff-d047e0f37f4b69c4818a0a79f15454b22933b6cc913d358b460a834e3ccc7f8dL58-R68))
*  Update the ieee754 dependency to 1.2.1 for the buffer package ([link](https://github.com/PipedreamHQ/pipedream/pull/6107/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL12888-R12886))
*  Update the string-width dependency to 4.2.3 for the cliui package ([link](https://github.com/PipedreamHQ/pipedream/pull/6107/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL24833-R24831))
*  Add a dev flag to the domexception package ([link](https://github.com/PipedreamHQ/pipedream/pull/6107/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR11635))
*  Remove the unused components/danny_test_app entry from the pnpm-lock.yaml file ([link](https://github.com/PipedreamHQ/pipedream/pull/6107/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL885-L887))

### CODE REVIEW

File: components/firebase_admin_sdk/actions/create-document/create-document.mjs [hunk 1]
```diff
@@ -24,11 +24,17 @@ export default {
         "data",
       ],
     },
+    customId: {
+      label: "Custom ID",
+      description: "Set this custom ID to the created document",
```
[the remark this answers is not in the document]
Comment (dylburger) [reply components/firebase_admin_sdk/actions/create-document/create-document.mjs:29]: Changed "to" to "for" to improve clarity

File: components/firebase_admin_sdk/actions/replicate-event-firestore/replicate-event-firestore.mjs [hunk 2]
```diff
@@ -5,7 +5,8 @@ import lodash from "lodash";
 export default {
   key: "firebase_admin_sdk-replicate-event-firestore",
   name: "Save Event to Firestore",
-  version: "0.4.2",
+  description: "Replicate event on firestore",
```
Comment (dylburger) [inline components/firebase_admin_sdk/actions/replicate-event-firestore/replicate-event-firestore.mjs:8]:
```suggestion
  description: "Replicate event in Firestore",
```

Comment (dylburger) [reply components/firebase_admin_sdk/actions/replicate-event-firestore/replicate-event-firestore.mjs:8]: Changed "on" to "in" to improve clarity

**Review by feyzullah — APPROVED** [review]
LGTM!

**Comment by vunguyenhung** [comment]
Hello everyone, I have tested this PR and there're some test cases failed or needed improvement. 

Please check the test report below for more information
[Firebase_6105_2150.pdf](https://github.com/PipedreamHQ/pipedream/files/11296821/Firebase_6105_2150.pdf)

File: components/firebase_admin_sdk/actions/create-document/create-document.mjs [hunk 3]
```diff
@@ -24,14 +24,20 @@ export default {
...
   },
   methods: {
     ...common.methods,
     async getResponse() {
-      return this.firebase.createDocument(this.collection, this.data);
+      return this.firebase.createDocument(this.collection, this.data, this.customId);
     },
     emitSummary($, response) {
-      $.export("$summary", `Successfully added document ${response._path.segments[1]}`);
+      $.export("$summary", `Successfully added document ${response?._path?.segments[1] ?? ""}`);
```
Comment (dylburger) [inline components/firebase_admin_sdk/actions/create-document/create-document.mjs:40]:
```suggestion
      $.export("$summary", `Successfully added document ${response?._path?.segments[1] ?? ""}`);
```

Comment (dylburger) [reply components/firebase_admin_sdk/actions/create-document/create-document.mjs:40]: Changed 'Successfully added document' to 'Successfully added the document' to follow Google's Developer docs style guide (https://developers.google.com/style/word-list#letter-S)

**Comment by vunguyenhung** [comment]
Hi everyone, all test cases are passed! Ready for release! 

Test report
[Firebase_6105_1417.pdf](https://github.com/PipedreamHQ/pipedream/files/11319779/Firebase_6105_1417.pdf)

**Review by feyzullah — APPROVED** [review]

**Pull request merged**